src/optimizer/plugins/CombineCodingAndQuantityValuesOptimizer.ts

```diff
@@ -137,3 +146,46 @@ export default {
     });
   }
 } as OptimizerPlugin;
+
+function moveUpCaretValueRule(
+  rule: ExportableCaretValueRule,
+  sd: ExportableProfile | ExportableExtension,
+  knownSiblings: ExportableCaretValueRule[]
+): void {
+  // if the caretPath ends with coding[0], and there are no sibling rules that use indices on coding,
+  // we can move the rule up a level to take advantage of how SUSHI handles rules on CodeableConcepts.
+  if (rule.caretPath.endsWith('coding[0]')) {
+    const basePath = rule.caretPath.replace(/\.coding\[0]$/, '');
+    const siblings = sd.rules.filter(
```

Comment:
I'd change this to:
```ts
const hasOtherSiblings = sd.rules.some(
```
Then in line 167 you'd have:
```ts
if (!hasOtherSiblings) {
```
To me, that's just a little more intentional -- and the use of `some` is more efficient than `filter` anyway (although efficiency isn't really a strong goal for this tool).

Reply:
BTW, same for 181/187 below.